backend/common/src/main/java/ai/verta/modeldb/common/authservice/AuthInterceptor.java

```diff
@@ -24,6 +26,21 @@
   public <R, S> Listener<R> interceptCall(
       ServerCall<R, S> call, Metadata requestHeaders, ServerCallHandler<R, S> next) {
     Context context = Context.current().withValue(METADATA_INFO, requestHeaders);
+
+    // To protect empty headers from request
+    Metadata.Key<String> email_key = Metadata.Key.of("email", Metadata.ASCII_STRING_MARSHALLER);
+    Metadata.Key<String> dev_key =
+        Metadata.Key.of("developer_key", Metadata.ASCII_STRING_MARSHALLER);
+    Metadata.Key<String> source_key = Metadata.Key.of("source", Metadata.ASCII_STRING_MARSHALLER);
+
+    if (requestHeaders == null
+        || !requestHeaders.containsKey(email_key)
+        || !requestHeaders.containsKey(dev_key)
+        || !requestHeaders.containsKey(source_key)) {
+      Status status = Status.newBuilder().setMessage("Required parameter is missing in metadata").build();
+      throw StatusProto.toStatusRuntimeException(status);
```

Comment:
I don't think we can throw exceptions from the interceptor

Reply:
Yes, agree i have fixed the code with the better solution